src/pages/search/index.tsx

```diff
@@ -1,12 +1,101 @@
 import { GetStaticPropsContext } from "next";
+import { useRouter } from "next/router";
+import { gql, useQuery } from "@apollo/client";
+import React from "react";
+import { LoadingSpinner, Koros } from "hds-react";
 
-import Page from "../../components/page/Page";
 import SearchPageSearchForm from "../../components/search/searchPageSearchForm/SearchPageSearchForm";
+import Page from "../../components/page/Page";
+import searchApolloClient from "../../api/searchApolloClient";
+import styles from "./search.module.scss";
+import { Connection, Item, Keyword, SearchResult } from "../../types";
+import { getNodes } from "../../api/utils";
+import List from "../../components/list/List";
+import Text from "../../components/text/Text";
+import Section from "../../components/section/Section";
+import SearchResultCard from "../../components/card/searchResultCard";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+export const SEARCH_QUERY = gql`
+  query SearchQuery($q: String) {
+    unifiedSearch(q: $q, index: "location", ontology: "Liikunta") {
+      edges {
+        node {
+          venue {
+            name {
+              fi
+            }
+            description {
+              fi
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+const emptyConnection = {
+  edges: [],
+};
+
+const mockKeywords: Keyword[] = [
+  { label: "Maauimala", onClick: noop },
+  { label: "Uinti", onClick: noop },
+  { label: "Ulkoliikuntapaikat", onClick: noop },
+];
+
+function getSearchResultsAsItems(
+  searchResultConnection: Connection<SearchResult> | null
+): Item[] {
+  const searchResults = getNodes<SearchResult>(searchResultConnection);
+  return searchResults.map((searchResult) => ({
+    id: searchResult.venue.name.fi,
+    title: searchResult.venue.name.fi,
+    infoLines: [],
+    href: "",
+    keywords: mockKeywords,
+    image:
+      "https://liikunta.hkih.production.geniem.io/uploads/sites/2/2021/05/097b0788-hkms000005_km00390n-scaled.jpeg",
+  }));
+}
 
 export default function Search() {
+  const {
+    query: { q: searchText },
+  } = useRouter();
+
+  const { data, loading } = useQuery(SEARCH_QUERY, {
+    client: searchApolloClient,
+    variables: { q: searchText ?? "*" },
+  });
+
+  const searchResultItems: Item[] = getSearchResultsAsItems(
+    data?.unifiedSearch ?? emptyConnection
+  );
+
   return (
     <Page title="Search" description="Search">
       <SearchPageSearchForm />
+      <Koros className={styles.koros} />
+      {loading ? (
+        <LoadingSpinner className={styles.spinner} />
+      ) : (
+        <Section variant="contained">
+          <Text
+            variant="h2"
+            className={styles.resultCount}
+          >{`${searchResultItems.length} hakutulosta`}</Text>
```

Comment:
Perhaps you should include a `role="status"` here

https://www.w3.org/WAI/WCAG21/working-examples/aria-role-status-searchresults/

I have a hunch that adding the attribute alone may not be enough. For instance, the result count may not get announced if the result count doesn't change. On the other hand, it's possible that the attribute works poorly with React and that you would need to consider some sort of aria-live package to be able to add proper support. This is a bit of a trial and error task where you would need to use Safari and VoiceOver to find out what works well enough.

Does TypeScript complain if you do:
```suggestion
          >{searchResultItems.length} hakutulosta</Text>
```